internal/server/storage/dbstorage.go

```diff
@@ -0,0 +1,225 @@
+package storage
+
+import (
+	"context"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/npavlov/go-metrics-service/internal/domain"
+	"github.com/npavlov/go-metrics-service/internal/server/db"
+	"github.com/npavlov/go-metrics-service/internal/server/dbmanager"
+)
+
+const maxRetries = 3
+
+type DBStorage struct {
+	Queries *db.Queries
+	l       *zerolog.Logger
+	dbCon   dbmanager.PgxPool
+}
+
+// NewDBStorage initializes a new DBStorage instance.
+func NewDBStorage(dbCon dbmanager.PgxPool, l *zerolog.Logger) *DBStorage {
```

Comment:
```suggestion
func NewDBStorage(dbCon dbmanager.PgxPool, logger *zerolog.Logger) *DBStorage {
```

Reply:
Where is our linter? :)

Reply:
fixed, it's interesting why I didn't get an error